### DISCUSSION TITLE
Constructing a rule to check a list of dictionaries?

### DISCUSSION THREAD

**Discussion by priyankmav** [zeroSteiner/rule-engine#16] [Q&A] [answered]
Hello zeroSteiner,

First things, thank you for this amazing python package!
I have a question regarding if a rule needs to be checked against a list of dictionaries.
For example below is the document -->
```
{
	"personFacts": [{
		"abc": false,
		"bcd": false,
		"edg": false,
		"asd": false,
		"effectiveTMS": {
			"$date": "1970-01-01T05:00:00.000Z"
		},
		"jjf": 0,
		"sha": false,
		"ncb": false,
		"qwe": false,
		"ytu": 0,
		"aops": 0,
		"countDaysCvrgEffTMS": 11,
		"api_response": [{
			"model": "surgery",
			"modelType": "xyz",
			"classifierValue": "high",

		}, {
			"model": "oncology",
			"modelType": "abc",
			"classifierValue": "low"
		}, {
			"model": "dermatology",
			"modelType": "cxx",
			"classifierValue": "medium"
		}]
	}],
	"createdTMS": {
		"$date": "2021-03-08T16:13:53.736Z"
	},
	"lastUpdatedTMS": {
		"$date": "2021-03-08T11:22:04.912Z"
	}
}
```

The rules initally were just checked for symbols/attributes in personFacts[0].
However, we have added a new attribute called the api_response to the member info.
The rule engine did a good job for simple rules like "bcd == false and effectiveTMS > $today" .
However, how would it be possible to construct a rule to access the api_response field as well?
I am looking at a rule something like this --> "bcd == false and any(model == 'oncology' in api_response)"

Could you help me here?

**Comment by zeroSteiner** [comment, marked as answer]
Well it's not currently possible, but I've already started to experiment with a solution that I think would address this problem nicely and that's supporting array comprehension expressions. This would be just like Python's list comprehension but in the Rule Engine syntax. You'd then make your rule like `bcd == false and [ item['model'] == 'oncology' for item in api_response ]` and it should work the way you've described.

Until that's done I can't really think of a solution. I also don't exactly have an ETA on when that will be done.

**Reply by priyankmav** [reply]
Yes, that solution looks neat. Shall wait for this feature then :)

**Reply by zeroSteiner** [reply]
Now that v3.2 has been released, this is possible using the example I provided.

**Reply by priyankmav** [reply]
Okay great! Thank you Spencer! :)

**Reply by MdTabraiz** [reply]
Hello zeroSteiner,

how can i add any or all condition  for a list of dictionaries.
something like this
$all( [ item['model'] == 'oncology' for item in api_response ])

**Reply by MdTabraiz** [reply]
when i am doing something like this i am getting the following error:
can not map python type 'ComprehensionExpression' to a compatible data type

**Reply by zeroSteiner** [reply]
~~@MdTabraiz that looks like a bug. Would you mind please opening an issue for it?~~

Nevermind, I was able to open #72 to track this.
